src/plugins/mer/mercmakebuildconfiguration.cpp

```diff
@@ -60,8 +60,13 @@ MerCMakeBuildConfiguration::MerCMakeBuildConfiguration(Target *target, Utils::Id
     connect(aspect, &BaseAspect::changed,
             this, &BuildConfiguration::updateCacheAndEmitEnvironmentChanged);
 
+    /*
+     * We want to be sure that BE is running before parsing starts.
+     * We should not start BE while parsing is in progress using AutoConnection - it lead to crash during kit disable.
+     * Instead we start BE on parsingStarted signal using QueuedConnection.
+     */
     connect(target, &Target::parsingStarted,
-            this, &MerCMakeBuildConfiguration::ensureBuildEngineRuns);
+            this, &MerCMakeBuildConfiguration::ensureBuildEngineRuns, Qt::QueuedConnection);
```

Comment:
The original code had the effect that the emitting side (where `emit parsingStarted()` is called) was blocked for the time of starting the build engine (which may involve dialog with the user who may or may not permit to start the build engine). With queued connection the emitting side continues immediately I believe, which is not wanted.

Now I checked the original PR and yeah I already mentioned the issue there.

We need to find some other solution, possibly with changes to cmake plugin, so that we can delay and possibly abort parsing depending on build engine readiness.

Reply:
> which may involve dialog with the user who may or may not permit to start the build engine

I did not notice that (maybe because I checked something like "always start VM and don't ask again" recently).

> With queued connection the emitting side continues immediately I believe, which is not wanted.

I will bring back start VM dialog and double check this.

> possibly with changes to cmake plugin

This condition may help, because I intentionally didn't touch QtC code while tried to fix this issue.